src/ramanchada2/misc/utils/matchsets.py

```diff
@@ -0,0 +1,227 @@
+
+
+def cost_matrix_peaks(spectrum_a_dict, spectrum_b_dict, threshold_max_distance=9, cost_func=None):
+    if cost_func is None:
+        cost_func = cost_function_position
+    peaks_a = np.array(list(spectrum_a_dict.keys()))
+    intensities_a = np.array(list(spectrum_a_dict.values()))
+    peaks_b = np.array(list(spectrum_b_dict.keys()))
+    intensities_b = np.array(list(spectrum_b_dict.values()))
+
+    num_peaks_b = len(peaks_b)  # Number of reference peaks to match
+
+    # Normalize intensities using min-max normalization
+    def normalize_intensities(intensities):
+        min_intensity = np.min(intensities)
+        max_intensity = np.max(intensities)
+        return (intensities - min_intensity) / (max_intensity - min_intensity)
+
+    intensities_a_normalized = normalize_intensities(intensities_a)
+    intensities_b_normalized = normalize_intensities(intensities_b)
+
+    num_peaks_a = len(peaks_a)
+    cost_matrix = np.full((num_peaks_a, num_peaks_b), np.inf)  # Initialize with infinity
+
+    for i in range(num_peaks_a):
+        for j in range(num_peaks_b):
+            cost = cost_func([peaks_a[i], intensities_a_normalized[i]], [peaks_b[j], intensities_b_normalized[j]], priority_weight=1)
+            cost_matrix[i, j] = cost
+    return cost_matrix
+
+
+def match_peaks(spectrum_a_dict, spectrum_b_dict, threshold_max_distance=9, df=False, cost_func=None):
+    """
+    Match peaks between two spectra based on their positions and intensities.
+
+    Uses scipy linear_sum_assignment to match peaks based on cost function
+    https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linear_sum_assignment.html
+
+    Parameters:
+    ----------
+    spectrum_a_dict : dict
+        A dictionary representing the first spectrum, where keys are peak
+        positions (float) and values are peak intensities (float).
+
+    spectrum_b_dict : dict
+        A dictionary representing the second spectrum, where keys are peak
+        positions (float) and values are peak intensities (float).
+
+    threshold_max_distance : float, optional
+        The maximum allowed distance for two peaks to be considered a match.
+        Default is 8.
+
+    df : bool, optional
+        If True, return a DataFrame with matched peaks and their respective
+        intensities; if False, return None
+
+    Returns:
+    -------
+    matched_peaks : (matched_peaks_a,matched_peaks_b,matched_distances, df)
+
+    Examples:
+    ---------
+    >>> spectrum_a = {100: 10, 105: 20, 110: 15}
+    >>> spectrum_b = {102: 12, 106: 22, 111: 16}
+    >>> match_peaks(spectrum_a, spectrum_b)
+
+    """
+    cost_matrix = cost_matrix_peaks(spectrum_a_dict, spectrum_b_dict, threshold_max_distance=threshold_max_distance, cost_func=cost_function if cost_func is None else cost_func)
+
+    # Use the Hungarian algorithm to find the optimal assignment
+    try:
+        row_ind, col_ind = linear_sum_assignment(cost_matrix)
```

Comment:
If we ignore the amplitude values `linear_sum_assignment` provides worse result than the already implemented `argmin2d`.

Consider this example:
```python
from scipy.optimize import linear_sum_assignment

import ramanchada2 as rc2
import numpy as np
import matplotlib.pyplot as plt

x = np.array([1, 1.5, 2, 2.5, 3, 6, 7, 7.1, 9, 10, 11, 12])
y = np.array([1, 2, 3, 4, 5, 6, 7, 8.])

lsa_x_idx, lsa_y_idx = np.array(linear_sum_assignment(np.abs(np.subtract.outer(x, y))))
gg_x_idx, gg_y_idx = rc2.misc.utils.find_closest_pairs_idx(x, y)

line_kw = dict(alpha=.5)
for i in x:
    plt.axvline(i, **line_kw)
    
for i in y:
    plt.axhline(i, **line_kw)
    
plt.plot(x[lsa_x_idx], y[lsa_y_idx], 'x', markersize=10, mew=3, label='linear_sum_assignmet')
plt.plot(x[gg_x_idx], y[gg_y_idx], '+', markersize=10, mew=3, label='gg argmin2d')
plt.legend(loc='lower right')
plt.xlabel('locations 1')
plt.ylabel('locations 2')
```